services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -327,13 +337,13 @@ public String toJsonPretty() {
 
     public static class DefaultBuilder implements EnhancedDocument.Builder {
 
-
         Map<String, AttributeValue> attributeValueMap = new LinkedHashMap<>();
         ChainConverterProvider converterProvider = ChainConverterProvider.create(DefaultAttributeConverterProvider.create());
         List<AttributeConverterProvider> attributeConverterProviders;
 
         public DefaultBuilder(DefaultEnhancedDocument enhancedDocument) {
-            attributeValueMap = enhancedDocument.attributeValueMap;
+            attributeValueMap = attributeValueMap != null ? new LinkedHashMap<>(enhancedDocument.attributeValueMap)
```

Comment:
Can `attributeValueMap` ever be null since it's instantiated at declaration time above?

Reply:
Yes , with the new implementation where attributeValueMap is passed in builder this value can be null